src/GameContext.js

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable no-console */
+import React, { createContext, useState, useEffect } from "react";
+import { io } from "socket.io-client";
+import { uniqueNamesGenerator, colors, animals } from "unique-names-generator";
+
+const socket = io("localhost:4000");
+// const socket = io("https://luzhanqi.herokuapp.com/");
+
+export const GameContext = createContext({});
+
+// eslint-disable-next-line react/prop-types
+export const GameProvider = ({ children }) => {
+  /** user-input: the user's name */
+  const defaultName = uniqueNamesGenerator({
+    dictionaries: [colors, animals],
+    length: 2,
+  });
+  const [playerName, setPlayerName] = useState(defaultName);
+  /** game ID assigned to host, or user-input: game Id entered by player */
+  const [roomId, setRoomId] = useState("");
+
+  const [host, setHost] = useState(false);
+  const [joinedGame, setJoinedGame] = useState(false);
+
+  const [clientTurn, setClientTurn] = useState(-1);
+  const [playerList, setPlayerList] = useState([]);
+  // eslint-disable-next-line no-unused-vars
+  const [myBoard, setMyBoard] = useState(Array(12).fill(Array(5).fill(null)));
+  const [myPositions, setMyPositions] = useState(
+    Array(6).fill(Array(5).fill(null))
+  );
+  const [submittedSide, setSubmittedSide] = useState(false);
+  const [pendingMove, setPendingMove] = useState({
+    source: [],
+    target: [],
+  });
+
+  /**
+   * Game phases:
+   * 0: waiting
+   * 1: placement
+   * 2: gameplay
+   */
+
+  // eslint-disable-next-line no-unused-vars
+  const [gamePhase, setGamePhase] = useState(0);
+
+  const boardPositions = {};
+
+  for (let j = 0; j < 6; j++) {
+    for (let i = 0; i < 5; i++) {
+      boardPositions[[j, i]] = "none";
+    }
+  }
+
+  const [startingBoard, setStartingBoard] = useState(boardPositions);
+
+  // const [startingBoard, setStartingBoard] = useState();
+  const [error, setError] = useState("");
+
+  const gameState = {
```

Comment:
Seems pretty solid. I would say that in the future we probably want to split this up into separate contexts, because some of the fields only belong to a specific stage of the game.

GameContext (data needed in all sub-components): playerName, roomId, host, error, ...
The setup component probably doesn't need a context because all of the data is localized.
LZQContext: board, selectedPiece, turn, pendingMove, timeLeft, ...
The results component probably doesn't need a context either.

 Also, when we implement reach router the `roomId` will be pulled directly from the query params.